src/components/VueDialogCard.vue

```diff
@@ -0,0 +1,146 @@
+<script lang="ts">
+import { defineComponent, computed } from 'vue';
+
+// types
+import { DialogCard } from 'components/types';
+
+// Component used to render dialog window which you can fill with custom content
+// Available slots:
+// - title
+// - metadata
+// - content
+// - buttons
+// - image
+// The content and image are rendered side by side in a scrollable dialog window
+export default defineComponent({
+  name: 'VueDialogCard',
+  props: {
+    dialog: {
+      type: Object as () => DialogCard,
+      required: true,
+    },
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  emits: ['update:modelValue', 'change'],
+  setup(props, { emit }) {
+    const modalOpened = computed({
+      get: () => props.modelValue,
+      set: (value) => {
+        emit('update:modelValue', value);
+        emit('change');
+      },
+    });
+
+    return {
+      modalOpened,
+    };
+  },
+});
+</script>
+
+<template>
```

Comment:
Comments are missing, please add them.

Reply:
Removed the historical file as it is replaced with DialogCard.